core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -811,9 +810,16 @@ public String toString(@Nullable AbstractBlockChain chain, @Nullable Network net
                         s.append('\n');
                     }
                 } catch (Exception e) {
```

Comment:
I might be helpful to know what type of exceptions are being caught here.  It's a large `try`-block and it is hard to know which operations might be causing exceptions and what the exceptions are.

I was able to change the type from `Exception` to `RuntimeException` and the code still compiles and passes tests and I think that change would be slightly helpful for people reading  the code. It might also be helpful to catch more specific exceptions and/or use multiple catch blocks to break down where in the code the exception might be occurring. As the code now stands there could be fatal exceptions that are ignored.

I realize this might be out-of-scope for this PR, but these issue make the PR harder to review.

Reply:
You're right, I hate these kind of big catch blocks too. But I don't know how to do it better in Java, as there is no "otherwise { }" (or similar). I thought the only line method call that can throw is the getScriptSig() call, so I now changed the catch to ScriptException. However, that uncovered an IndexOutOfBounds in a test…

…leading to this bugfix, which we'll need to merge first: #3635

Reply:
An alternative to the catch block would be a "isScriptSigValid()" that the getScriptSig() could be guarded with, but that would lead to everything parsed twice.